extensions/skins/conversation_v1/Conversation.js

```diff
@@ -75,227 +218,425 @@ oppia.directive('conversationSkin', [function() {
+          return true;
+        } else {
+          return panelName == $scope.currentVisiblePanelName;
+        }
+      };
 
-          var page = $('html, body, iframe');
+      $scope.setVisiblePanel = function(panelName) {
+        $scope.currentVisiblePanelName = panelName;
+        if (panelName === $scope.PANEL_INTERACTION) {
+          $scope.$broadcast('showInteraction');
+        }
+      };
+
+      $scope.resetVisiblePanel = function() {
+        if ($scope.panels.length === 0) {
+          $scope.currentVisiblePanelName = null;
+        } else {
+          $scope.currentVisiblePanelName = $scope.panels[0];
+        }
+      };
 
-          page.on("scroll mousedown wheel DOMMouseScroll mousewheel keyup touchmove", function() {
-            page.stop();
-          });
+      // Changes the currently-active card, and resets the 'show previous
+      // responses' setting.
+      var _navigateToCard = function(index) {
+        $scope.activeCard = $scope.transcript[index];
+        $scope.arePreviousResponsesShown = false;
+        $scope.interactionIsInline = oppiaPlayerService.isInteractionInline(
+          $scope.activeCard.stateName);
 
-          page.animate({
-            'scrollTop': newScrollTop
-          }, 1000, 'easeOutQuad', function() {
-            page.off("scroll mousedown wheel DOMMouseScroll mousewheel keyup touchmove");
-          }).promise().done(postScrollCallback);
-        });
+        _recomputeAndResetPanels();
       };
 
-      var _addNewCard = function(stateName, contentHtml) {
-        $scope.allResponseStates.push({
+      var _addNewCard = function(stateName, contentHtml, interactionHtml) {
+        var interactionIsInline = oppiaPlayerService.isInteractionInline(
+          stateName);
+        var interactionInstructions = (
+          oppiaPlayerService.getInteractionInstructions(stateName));
+
+        var oldStateName = null;
+        if ($scope.transcript.length >= 1) {
+          oldStateName = $scope.transcript[$scope.transcript.length - 1].stateName;
+        }
+
+        oppiaPlayerService.applyCachedParamUpdates();
+        $scope.transcript.push({
           stateName: stateName,
           content: contentHtml,
+          interaction: interactionHtml,
+          interactionIsInline: interactionIsInline,
+          interactionIsDisabled: false,
+          interactionInstructions: interactionInstructions,
           answerFeedbackPairs: []
         });
+
+        $scope.numProgressDots++;
+
+        if ($scope.canWindowFitTwoCards() &&
+            oldStateName &&
+            oppiaPlayerService.isInteractionInline(oldStateName) &&
+            !oppiaPlayerService.isInteractionInline(stateName)) {
+          $scope.isAnimatingToTwoCards = true;
+          $timeout(function() {
+            $scope.isAnimatingToTwoCards = false;
+            $scope.currentProgressDotIndex = $scope.numProgressDots - 1;
+          }, TIME_NUM_CARDS_CHANGE_MSEC + TIME_FADEIN_MSEC + 500);
```

Comment:
Unsure. This (and the following magic numbers) are essentially just padding -- they weren't chosen systematically. Giving it a name feels like it exaggerates its importance; on the other hand, there's some value in naming constants when it makes sense.

I lean mildly towards not naming and leaving as-is, but what do you think?